web/bitwarden/ciphers.go

```diff
@@ -529,6 +535,82 @@ func DeleteCipher(c echo.Context) error {
 	return c.NoContent(http.StatusOK)
 }
 
+// SoftDeleteCipher is the handler for the route to soft delete a cipher.
+// See https://github.com/bitwarden/server/pull/684 for the bitwarden implementation
+func SoftDeleteCipher(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, permission.PUT, consts.BitwardenCiphers); err != nil {
+		return c.JSON(http.StatusUnauthorized, echo.Map{
+			"error": "invalid token",
+		})
+	}
+
+	id := c.Param("id")
+	if id == "" {
+		return c.JSON(http.StatusNotFound, echo.Map{
+			"error": "missing id",
+		})
+	}
+
+	cipher := &bitwarden.Cipher{}
+	if err := couchdb.GetDoc(inst, consts.BitwardenCiphers, id, cipher); err != nil {
+		if couchdb.IsNotFoundError(err) {
+			return c.JSON(http.StatusNotFound, echo.Map{
+				"error": "not found",
+			})
+		}
+		return c.JSON(http.StatusInternalServerError, echo.Map{
+			"error": err.Error(),
+		})
+	}
+	cipher.Metadata.ChangeUpdatedAt()
+	cipher.DeletedDate = &cipher.Metadata.UpdatedAt
+	if err := couchdb.UpdateDoc(inst, cipher); err != nil {
+		return c.JSON(http.StatusInternalServerError, echo.Map{
+			"error": err.Error(),
+		})
+	}
```

Comment:
I think we should also update the revision date like for UpdateCipher:

```go
_ = settings.UpdateRevisionDate(inst, setting)
```

Reply:
Done with e4de808